test/new-e2e/shared-components/configcheck_test.go

```diff
@@ -0,0 +1,196 @@
+		Settings:   fmt.Sprintf("%s", matches[settingsIndex:]), // format to string for assertion
+	}, nil
+}
+
+func (v *agentSuite) TestMatchToTemplateHelper() {
+	sampleCheck := `=== uptime check ===
+Configuration provider: file
+Configuration source: file:/etc/datadog-agent/conf.d/uptime.d/conf.yaml.default
+Config for instance ID: uptime:c72f390abdefdf1a
+key: value
+path: http://example.com/foo
+~
+===
+
+=== npt check ===
+Configuration provider: file
+Configuration source: file:/etc/datadog-agent/conf.d/npt.d/conf.yaml.default
+Config for instance ID: npt:c72f390abdefdf1a
+{}
+~
+===`
+
+	result, err := MatchCheckToTemplate("uptime", sampleCheck)
+	assert.NoError(v.T(), err)
+
+	assert.Contains(v.T(), result.CheckName, "uptime")
+	assert.Contains(v.T(), result.Filepath, "file:/etc/datadog-agent/conf.d/uptime.d/conf.yaml.default")
+	assert.Contains(v.T(), result.InstanceId, "uptime:c72f390abdefdf1a")
+	assert.Contains(v.T(), result.Settings, "key: value")
+	assert.Contains(v.T(), result.Settings, "path: http://example.com/foo")
+	assert.NotContains(v.T(), result.Settings, "{}")
+}
+
+// cpu, disk, file_handle, io, load, memory, network, ntp, uptime
+func (v *agentSuite) TestDefaultInstalledChecks() {
+	v.UpdateEnv(e2e.AgentStackDef(nil))
+
+	testChecks := []CheckConfigOutput{
+		CheckConfigOutput{
+			CheckName:  "cpu",
+			Filepath:   "file:/etc/datadog-agent/conf.d/cpu.d/conf.yaml.default",
+			InstanceId: "cpu:",
+			Settings:   "{}",
+		},
+		CheckConfigOutput{
+			CheckName:  "disk",
+			Filepath:   "file:/etc/datadog-agent/conf.d/disk.d/conf.yaml.default",
+			InstanceId: "disk:",
+			Settings:   "use_mount: false",
+		},
+		CheckConfigOutput{
+			CheckName:  "file_handle",
+			Filepath:   "file:/etc/datadog-agent/conf.d/file_handle.d/conf.yaml.default",
+			InstanceId: "file_handle:",
+			Settings:   "{}",
+		},
+		CheckConfigOutput{
+			CheckName:  "io",
+			Filepath:   "file:/etc/datadog-agent/conf.d/io.d/conf.yaml.default",
+			InstanceId: "io:",
+			Settings:   "{}",
+		},
+		CheckConfigOutput{
+			CheckName:  "load",
+			Filepath:   "file:/etc/datadog-agent/conf.d/load.d/conf.yaml.default",
+			InstanceId: "load:",
+			Settings:   "{}",
+		},
+		CheckConfigOutput{
+			CheckName:  "memory",
+			Filepath:   "file:/etc/datadog-agent/conf.d/memory.d/conf.yaml.default",
+			InstanceId: "memory:",
+			Settings:   "{}",
+		},
+		CheckConfigOutput{
+			CheckName:  "network",
+			Filepath:   "file:/etc/datadog-agent/conf.d/network.d/conf.yaml.default",
+			InstanceId: "network:",
+			Settings:   "{}",
+		},
+		CheckConfigOutput{
+			CheckName:  "ntp",
+			Filepath:   "file:/etc/datadog-agent/conf.d/ntp.d/conf.yaml.default",
+			InstanceId: "ntp:",
+			Settings:   "{}",
+		},
+		CheckConfigOutput{
+			CheckName:  "uptime",
+			Filepath:   "file:/etc/datadog-agent/conf.d/uptime.d/conf.yaml.default",
+			InstanceId: "uptime:",
+			Settings:   "{}",
+		},
+	}
+
+	output := v.Env().Agent.ConfigCheck()
+	fmt.Println(output)
```

Comment:
**❓ question:** ‏do we want to print this or is it a debug leftover ?

Reply:
😅 debugging output, I will clean it up

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/18290/commits/626478c9dc277ccd0568ca6f9a37a779b881cc3b